services/scheduler.go

```diff
@@ -159,8 +159,11 @@ func (ot *OneTime) RunJobAt(initr models.Initiator, job models.JobSpec) {
 	select {
 	case <-ot.done:
 	case <-ot.Clock.After(initr.Time.DurationFromNow()):
-		_, err := BeginRun(job, initr, models.RunResult{}, ot.Store)
-		if err != nil {
+		if err := ot.Store.MarkRan(&initr); err != nil {
+			logger.Error(err.Error())
+			return
+		}
+		if _, err := BeginRun(job, initr, models.RunResult{}, ot.Store); err != nil {
```

Comment:
I'm mulling these scenarios over in my head and wanted them to be explicitly listed so we're on the same page:

1. Mark as ran and then chainlink node shuts down before `BeginRun`, we'll lose the run [edge case].
2. Mark as ran and then `BeginRun` errors, we have an errored ran run.

Are there ways to prevent these situations or are they tolerable? @se3000

Reply:
Might have to set ran after it completes. to prevent those.

Reply:
The problem with setting Ran afterwards is it opens up a race condition, where two runs happen simultaneously. Maybe worrying about that is overly cautious... Between the current design of the code, and the lock on the database, the race condition scenario should never happen, but I'm inclined to try to guard against it now, to prevent against something sneaking in in the future.

1: Doesn't the scheduler gracefully shut down? It should wait for the job runs to finish. Or are you talking about like killing the process? Killing the process seems like a bit of an edge case, but something we should keep in mind and circle back to fix in a subsequent branch.

2: I guess that depends on where it errors. 
2a) If it errors out somewhere before entering `services.ExecuteRun`, then I'm leaning towards rolling back the `Ran` to false, because the Run itself didn't error, just some part of the node(saving to the db, logger... things that should really never go wrong.)
2b) If the Run errors(like returns with status "errored"), that's perfectly valid, no need to revert the `Ran` flag.